### ISSUE TITLE
Can a custom control be used with multiple maps?

### ISSUE THREAD

**Issue by umbe1987** [Viglino/ol-ext-angular#7]
Sorry to bother again...
hope you can help, otherwise, please feel free to ignore this message.

I am interested in adding a custom control outside my maps as you did with `app-mouse-position`.
My goal is to have a single control that can be used with multiple maps to (e.g.) list the number of layers wthin the clicked map somewhere. This control will be placed in a target outside the maps.

However, am I supposed to add it to use [`setMap()`](https://openlayers.org/en/latest/apidoc/module-ol_control_Control-Control.html#setMap) upon creation, or shall I simply define it and use it when I need it?

I see from your `app-component.html` that you define your [`app-mouse-postion`](https://github.com/Viglino/ol-ext-angular/blob/3d0c5950e22e459fbdae94ff2afea086a8dceaab/src/app/app.component.html#L29-L31) twice, once with a `mapId` attribute and once without. May I ask you why? I would be interested in how the one without `mapId` could be used for my case.
```
<!-- Define control outside the map -->
<app-mouse-position></app-mouse-position>
<app-mouse-position mapId="map1"></app-mouse-position>
```
I think my main problem is that I will have to switch map in my control with `setMap()` every time I click on a different map, while in your example I guess the mouse coordinates can be taken from either of the two maps as they are in sync, am I correct?

**Comment by Viglino** [comment]
Hello,

A control is associated with only **one** map. So you have to create one control per map (as mouse position in the example).
The trick is to hide the control when the map is not concerned and show only one at a time.

You can use `setMap` to switch the control map when a map is active.
But don't have to use it, you can add a control per map and you just have to show / hide the control when the map is active or not (if the mouse is on the viewport or if a map is clicked).

@+

**Comment by umbe1987** [comment]
Thank you so much again for your precious feedback!
I was able to implement the outside controls.
What I did was:

- create one control per each map and associate each of them to one map
- draw them concurrently BUT one on top of the other, to give the impression that it's just one control
- create a service that, after clicking one control (they actually are all buttons in my case), handles the activation/deactivation of all other similar controls underneath
- activation/deactivation of controls also implies adding/removing the click event listener to each maps the controls are associated with (i.e. by using e.g. `this.clickKey = map.on('click', whatever)` to a variable on activation and using `unByKey(this.clickKey)` when deactivating

Again, thankyou so much for clatrifying the 1 control -> 1 map rule, and to give me a suggestion for solving the problem!

Closing now :+1:
